spacer/tests/test_extract_features.py

```diff
@@ -177,5 +178,133 @@ def test_regression(self):
+                                   key='kh6dydiix0.jpeg',
+                                   bucket_name='spacer-test'),
+            feature_loc=DataLocation(storage_type='memory',
+                                     key='dummy')
+        )
+        ext = feature_extractor_factory(msg.feature_extractor_name)
+        img = load_image(msg.image_loc)
+        features, return_msg = ext(img, msg.rowcols)
+
+        self.assertTrue(isinstance(return_msg, ExtractFeaturesReturnMsg))
+        self.assertTrue(isinstance(features, ImageFeatures))
+
+        # Check some feature metadata
+        self.assertEqual(features.point_features[0].row, 148)
+        self.assertEqual(features.point_features[0].col, 50)
+        self.assertEqual(len(features.point_features[0].data), 1280)
+
+    def test_corner_case2(self):
+        msg = ExtractFeaturesMsg(
+            job_token='cornercase_2',
+            feature_extractor_name='efficientnet_b0_ver1',
+            rowcols=[(190, 226), (25, 359)],
+            image_loc=DataLocation(storage_type='s3',
+                                   key='sfq2mr5qbs.jpeg',
+                                   bucket_name='spacer-test'),
+            feature_loc=DataLocation(storage_type='memory',
+                                     key='dummy')
+        )
+
+        ext = feature_extractor_factory(msg.feature_extractor_name)
+        img = load_image(msg.image_loc)
+        features, return_msg = ext(img, msg.rowcols)
+
+        self.assertTrue(isinstance(return_msg, ExtractFeaturesReturnMsg))
+        self.assertTrue(isinstance(features, ImageFeatures))
+
+        # Check some feature metadata
+        self.assertEqual(features.point_features[0].row, 190)
+        self.assertEqual(features.point_features[0].col, 226)
+        self.assertEqual(len(features.point_features[0].data), 1280)
+
+    def test_regression(self):
+        rowcols = [(20, 265),
+                   (76, 295),
+                   (59, 274),
+                   (151, 62),
+                   (265, 234)]
+
+        msg = ExtractFeaturesMsg(
+            job_token='regression_job',
+            feature_extractor_name='efficientnet_b0_ver1',
+            rowcols=rowcols,
+            image_loc=DataLocation(storage_type='s3',
+                                   key='08bfc10v7t.png',
+                                   bucket_name='spacer-test'),
+            feature_loc=DataLocation(storage_type='memory',
+                                     key='dummy')
+        )
+
+        legacy_feat_loc = DataLocation(storage_type='s3',
+                                       key='08bfc10v7t.png.effnet.featurevector',
+                                       bucket_name='spacer-test')
+
+        ext = feature_extractor_factory(msg.feature_extractor_name)
+
+        img = load_image(msg.image_loc)
+        features_new, _ = ext(img, msg.rowcols)
+        features_legacy = ImageFeatures.load(legacy_feat_loc)
+
+        self.assertFalse(features_legacy.valid_rowcol)
```

Comment:
This is confusing. Why is `valid_rowcol = False`? Didn't you extract these features using this code-base? If so, the ImageFeatures dataclass should be stored with valid rowcols. The legacy thing is just for the old extractor deployed on CoralNet..

Reply:
no, I didn't use this code-base to extract the legacy features, instead, I used the one in [CoralNet](https://github.com/qiminchen/CoralNet/blob/f0c5985cdc812ec8e42a880035fa5961a8f894d6/extract.py#L82) repo, I was thinking if we use the same code to extract features, the features are guaranteed to be the same without concerning `e.g. libjpeg` issue, though we know the new extractor is correct as the regression test work so well. what do you think?

Reply:
I see. So the legacy is from your own codebase. I like that. So perhaps we just try with `pngs` to rule out the libjpeg issue during the tests?